File: backend/src/main/java/com/pofo/backend/domain/comment/dto/response/CommentDetailResponse.java
package com.pofo.backend.domain.comment.dto.response;

import com.pofo.backend.domain.comment.entity.Comment;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
public class CommentDetailResponse {

    private Long id;
    private String content;
    private LocalDateTime createdAt;
    private final String type = "comment";

    public static CommentDetailResponse from(Comment comment) {
        return new CommentDetailResponse(comment.getId(), comment.getContent(), comment.getCreatedAt());
    }
}

File: backend/src/main/java/com/pofo/backend/domain/comment/dto/response/CommentDetailResponse.kt
package com.pofo.backend.domain.comment.dto.response

import com.pofo.backend.domain.comment.entity.Comment
import java.time.LocalDateTime

data class CommentDetailResponse(
    val id: Long?,
    val content: String,
    val createdAt: LocalDateTime,
    val type: String = "comment"
) {
    companion object {
        fun from(comment: Comment): CommentDetailResponse {
            return CommentDetailResponse(
                id = comment.id,
                content = comment.content,
                createdAt = comment.createdAt
            )
        }
    }
}